### PR TITLE
Use assertArrayEquals() to see reported markers on flakey test

### PR DESCRIPTION
**Pull request by briandealwis**

### CODE REVIEW

**Comment by chanseokoh** [comment]
I actually emulated a test failure, and unfortunately, this doesn't provide any more useful info:

```
java.lang.AssertionError: array lengths differed, expected.length=2 actual.length=0
	at org.junit.Assert.fail(Assert.java:88)
	...
```

**Comment by briandealwis** [comment]
I should have tested.  I've written an alternative.
screen shot 2017-06-16 at 10 51 17 pm

File: plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/XsltSourceQuickFixTest.java [hunk 1]
```diff
@@ -58,7 +61,7 @@
   public void testApply() throws CoreException {
 
     IProject project = appEngineStandardProject.getProject();
-    IFile file = project.getFile("appengine-web.xml");
+    final IFile file = project.getFile("appengine-web.xml");
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/XsltSourceQuickFixTest.java]: No need for `final`.

**Comment by briandealwis** [comment]
Failure 1 (Neon: [full log](https://api.travis-ci.org/jobs/244557052/log.txt?deansi=true)):
```
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 25.888 sec <<< FAILURE! - in com.google.cloud.tools.eclipse.appengine.validation.XsltSourceQuickFixTest
testApply(com.google.cloud.tools.eclipse.appengine.validation.XsltSourceQuickFixTest)  Time elapsed: 25.687 sec  <<< FAILURE!
java.lang.AssertionError: [/test0.06296464251836487/appengine-web.xml:2: Project ID should be specified at deploy time]
	at org.junit.Assert.fail(Assert.java:88)
	at com.google.cloud.tools.eclipse.test.util.ArrayAssertions.assertIsEmpty(ArrayAssertions.java:58)
	at com.google.cloud.tools.eclipse.test.util.ArrayAssertions.assertIsEmpty(ArrayAssertions.java:36)
	at com.google.cloud.tools.eclipse.appengine.validation.XsltSourceQuickFixTest.testApply(XsltSourceQuickFixTest.java:92)
```

Failure 2 was on Oxygen: the job succeeded, but Travis killed the job as it went to 50 mins as it was uploading the build cache file.

**Comment by briandealwis** [comment]
I wondered if `ProjectUtils.waitForProjects()` may not actually wait for our validation checks, but it does.
```
Thread [main] (Suspended)	
	owns: RunnableLock  (id=224)	
	waiting for: Semaphore  (id=229)	
	Object.wait(long) line: not available [native method]	
	Semaphore.acquire(long) line: 39	
	JobManager.join(InternalJob, long, IProgressMonitor) line: 909	
	ValBuilderJob(InternalJob).join() line: 341	
	ValBuilderJob(Job).join() line: 458	
	ProjectUtils.waitForProjects(Runnable, IProject...) line: 209	
	ProjectUtils.waitForProjects(IProject...) line: 190	
	TestProjectCreator.addFacets() line: 172	
	TestProjectCreator.createProject(String) line: 142	
	TestProjectCreator.createProjectIfNecessary() line: 126	
	TestProjectCreator.getProject() line: 113	
	TestProjectCreator.getFacetedProject() line: 118	
	JavaFacetChangeTests.testChangeToAndFrom_1_8() line: 54	
[…]
Thread [Worker-8] (Suspended (entry into method checkForElements in AppEngineWebXmlValidator))	
	AppEngineWebXmlValidator.checkForElements(IResource, Document) line: 32	
	XmlValidator.validate(IFile, byte[]) line: 89	
	XmlValidator.validate(ValidationEvent, ValidationState, IProgressMonitor) line: 72	
	Validator$V2.validate(IResource, int, ValOperation, IProgressMonitor, ValidationEvent) line: 1158	
	ValManager.validate(Validator, ValOperation, IResource, int, IProgressMonitor, ValidationEvent) line: 704	
	ValManager$1.visit(Validator, IProject, ValType, ValOperation, IProgressMonitor) line: 665	
	ValManager.accept(IValidatorVisitor, IProject, IResource, ValType, ValOperation, IProgressMonitor) line: 810	
	ValManager.validate(IProject, IResource, int, ValType, int, ValOperation, IProgressMonitor) line: 669	
	ValBuilderJob$Visitor.visit(IResource) line: 299	
	Resource$2.visit(IResourceProxy) line: 119	
	Resource$1.visitElement(ElementTree, IPathRequestor, Object) line: 83	
	ElementTreeIterator.doIteration(DataTreeNode, IElementContentVisitor) line: 82	
	ElementTreeIterator.doIteration(DataTreeNode, IElementContentVisitor) line: 87	
	ElementTreeIterator.doIteration(DataTreeNode, IElementContentVisitor) line: 87	
	ElementTreeIterator.doIteration(DataTreeNode, IElementContentVisitor) line: 87	
	ElementTreeIterator.iterate(IElementContentVisitor) line: 129	
	Project(Resource).accept(IResourceProxyVisitor, int, int) line: 93	
	Project(Resource).accept(IResourceProxyVisitor, int) line: 51	
	Project(Resource).accept(IResourceVisitor, int, int) line: 116	
	Project(Resource).accept(IResourceVisitor) line: 104	
	ValBuilderJob.fullBuild(ValBuilderJob$ValidationRequest, IProgressMonitor) line: 219	
	ValBuilderJob.run(ValBuilderJob$ValidationRequest, IProgressMonitor) line: 178	
	ValBuilderJob.runInWorkspace(IProgressMonitor) line: 126	
	ValBuilderJob(InternalWorkspaceJob).run(IProgressMonitor) line: 39	
	Worker.run() line: 55	

```

**Comment by briandealwis** [comment]
Committed as it helps in diagnosing #2076.

**Pull request merged**